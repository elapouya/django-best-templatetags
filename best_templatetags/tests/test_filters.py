from django.test import TestCase
from django.template.loader import render_to_string
import os.path

TEST_DIR = os.path.dirname(os.path.realpath(__file__))

class FiltersTestCase(TestCase):
    compare_file = 'test_filters.out'
    maxDiff = None

    def setUp(self):
        self.context = dict(
            file_path='/a/b/c/myfile.extension',
            val50=50,
            valstar='*',
        )

    def test_filters(self):
        reference_filename = os.path.join(TEST_DIR, self.compare_file)
        rendered = render_to_string('best_templatetags/test_filters.html', self.context)
        with open(reference_filename) as fh:
            reference = fh.read()
        self.assertMultiLineEqual(reference,rendered)

    def generate_ref_file(self):
        reference_filename = os.path.join(TEST_DIR, self.compare_file)
        rendered = render_to_string('best_templatetags/test_filters.html', self.context)
        with open(reference_filename,'w') as fh:
            fh.write(rendered)