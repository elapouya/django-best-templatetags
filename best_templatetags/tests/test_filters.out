
basename
========

With file_path = /a/b/c/myfile.extension
{{ file_path|basename }} = myfile.extension

dirname
=======

With file_path = /a/b/c/myfile.extension
{{ file_path|dirname }} = /a/b/c

multiply
========
With val50 = 50
{{ val50|multiply:1024 }} = 51200

With valstar = '*'
{{ valstar|multiply:5 }} = *****
